File: web/remove_bg.py
from PIL import Image
import sys

def make_white_transparent(image_path, output_path, tolerance=240):
    try:
        img = Image.open(image_path).convert("RGBA")
        datas = img.getdata()

        newData = []
        for item in datas:
            # If all RGB values are above the tolerance, consider it white
            if item[0] > tolerance and item[1] > tolerance and item[2] > tolerance:
                # Keep color, set alpha to 0
                newData.append((item[0], item[1], item[2], 0))
            else:
                newData.append(item)

        img.putdata(newData)
        img.save(output_path, "PNG")
        print(f"Successfully processed {image_path} and saved to {output_path}")
    except Exception as e:
        print(f"Error processing image: {e}")
        sys.exit(1)

if __name__ == "__main__":
    make_white_transparent("c:/Users/ASUS/Downloads/ayatech-site-main/ayatech-site-main/web/public/logo_v3.png", "c:/Users/ASUS/Downloads/ayatech-site-main/ayatech-site-main/web/public/logo_transparent.png", 250)

File: web/remove_bg.js
const Jimp = require('jimp');

Jimp.read('c:/Users/ASUS/Downloads/ayatech-site-main/ayatech-site-main/web/public/logo_v3.png')
  .then(image => {
    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
      var red = this.bitmap.data[idx + 0];
      var green = this.bitmap.data[idx + 1];
      var blue = this.bitmap.data[idx + 2];
      
      // If color is close to white
      if (red > 235 && green > 235 && blue > 235) {
        this.bitmap.data[idx + 3] = 0; // Make transparent
      }
    });
    image.write('c:/Users/ASUS/Downloads/ayatech-site-main/ayatech-site-main/web/public/logo_v3_transparent.png');
    console.log("Logo updated");
  })
  .catch(err => {
    console.error(err);
  });
